Automattic/jetpack
# Fix/minor my jetpack improvement

Fixes #
- Improves the plan check on the boost card by using the new hasPaidPlanForProduct property on the product
- Adds a conditional check in the post-activation logic

## Proposed changes:
* This diff just makes a few small logic improvements

## Testing instructions:

* Navigate to My Jetpack and connect
* After the boost speed score bar generates on the my Jetpack card, confirm the tooltip that shows for the score matches one of the tooltips used when no plan is present
* Get a paid boost plan for the site, confirm that the tooltip on the score updates to used the paid language
